src/System.IdentityModel.Tokens.Jwt/JwtHeader.cs

```diff
@@ -102,6 +102,43 @@ public JwtHeader(SigningCredentials signingCredentials, IDictionary<string,strin
             SigningCredentials = signingCredentials;
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="JwtHeader"/>.
+        /// With the Header Parameters:
+        /// <para>{ { typ, JWT }, { alg, SigningCredentials.Algorithm } }</para>
+        /// </summary>
+        /// <param name="signingCredentials"><see cref="SigningCredentials"/> used when creating a JWS Compact JSON.</param>
+        /// <param name="outboundAlgorithmMap">provides a mapping for the 'alg' value so that values are within the JWT namespace.</param>
+        /// /// <param name="tokenType"> provides the token type</param>
+        public JwtHeader(SigningCredentials signingCredentials, IDictionary<string, string> outboundAlgorithmMap, string tokenType)
+            : base(StringComparer.Ordinal)
+        {
+            if (signingCredentials == null)
+            {
+                this[JwtHeaderParameterNames.Alg] = SecurityAlgorithms.None;
+            }
+            else
+            {
+                if (outboundAlgorithmMap != null && outboundAlgorithmMap.TryGetValue(signingCredentials.Algorithm, out string outboundAlg))
+                    Alg = outboundAlg;
+                else
+                    Alg = signingCredentials.Algorithm;
+
+                if (!string.IsNullOrEmpty(signingCredentials.Key.KeyId))
+                    Kid = signingCredentials.Key.KeyId;
+
+                if (signingCredentials is X509SigningCredentials x509SigningCredentials)
+                    this[JwtHeaderParameterNames.X5t] = Base64UrlEncoder.Encode(x509SigningCredentials.Certificate.GetCertHash());
+            }
```

Comment:
everything here and above looks the same as the ctor directly above.

Can you call that one, and then only keep the typ setting logic here?